TOOLS/PERF: Enable specifying different memory types for sender and receiver

souravzzz opened the pull request:
## What
Add support for specifying different memory types as source and destination in perftest

## Why ?
Currently the same memory type is used by the sender and the receiver.
Hence, only H-H and D-D transfers can be tested.
This PR adds support for H-D and D-H transfers.
Backward compatibility is preserved by using the same memory type as source and destination if only one is specified.

swx-jenkins3 commented:
Can one of the admins verify this patch?

souravzzz commented:
Example:

```
$ ./bin/ucx_perftest localhost -f -c 2 -t tag_lat -m rocm
+--------------+-----------------------------+---------------------+-----------------------+
|              |       latency (usec)        |   bandwidth (MB/s)  |  message rate (msg/s) |
+--------------+---------+---------+---------+----------+----------+-----------+-----------+
| # iterations | typical | average | overall |  average |  overall |   average |   overall |
+--------------+---------+---------+---------+----------+----------+-----------+-----------+
       1000000     1.433     7.848    11.025       0.97       0.69      127413       90702
$ ./bin/ucx_perftest localhost -f -c 2 -t tag_lat -m host,rocm
+--------------+-----------------------------+---------------------+-----------------------+
|              |       latency (usec)        |   bandwidth (MB/s)  |  message rate (msg/s) |
+--------------+---------+---------+---------+----------+----------+-----------+-----------+
| # iterations | typical | average | overall |  average |  overall |   average |   overall |
+--------------+---------+---------+---------+----------+----------+-----------+-----------+
       1000000     0.792     2.302     2.338       3.31       3.26      434408      427652
$ ./bin/ucx_perftest localhost -f -c 2 -t tag_lat -m rocm,host
+--------------+-----------------------------+---------------------+-----------------------+
|              |       latency (usec)        |   bandwidth (MB/s)  |  message rate (msg/s) |
+--------------+---------+---------+---------+----------+----------+-----------+-----------+
| # iterations | typical | average | overall |  average |  overall |   average |   overall |
+--------------+---------+---------+---------+----------+----------+-----------+-----------+
       1000000     0.775     2.284     2.292       3.34       3.33      437768      436257
```

dmitrygx commented:
ok to test

dmitrygx commented on src/tools/perf/perftest.c:
```diff
@@ -274,6 +275,9 @@ static void print_header(struct perftest_context *ctx)
                 test_api_str = "protocol layer";
                 test_data_str = "(automatic)"; /* TODO contig/stride/stream */
+                ucs_snprintf_zero(test_mem_str, sizeof(test_mem_str), "%s,%s",
+                        ucs_memory_type_names[ctx->params.send_mem_type],
```
pls align with the line above

dmitrygx commented on src/tools/perf/perftest.c:
```diff
@@ -282,6 +286,7 @@ static void print_header(struct perftest_context *ctx)
             printf("| API:          %-60s               |\n", test_api_str);
             printf("| Test:         %-60s               |\n", test->desc);
             printf("| Data layout:  %-60s               |\n", test_data_str);
+            printf("| Memory types: %-60s               |\n", test_mem_str);
```
why not print `ucs_memory_type_names[ctx->params.send_mem_type]` and `ucs_memory_type_names[ctx->params.recv_mem_type]` directly here? So, `test_mem_str` can be removed in this case
UCT perftest also supports different memtypes

souravzzz replied:
We want to print both the send and recv memory types. Since these two strings can of different lengths, making sure that the print is properly aligned to 60 characters becomes complicated.

dmitrygx replied:
maybe better separate it to:
```
printf("| Send memory type    : %-60s               |\n", ucs_memory_type_names[ctx->params.send_mem_type]);
printf("| Receive memory type: %-60s               |\n", ucs_memory_type_names[ctx->params.recv_mem_type]);
```

dmitrygx replied:
@souravzzz wdyt about separating them?

souravzzz replied:
Wanted to keep the print similar to that of data layout, which also prints the src and the dst types. But I agree this makes is more clear.

dmitrygx commented on src/tools/perf/perftest.c:
```diff
@@ -460,6 +466,31 @@ static ucs_status_t parse_ucp_datatype_params(const char *optarg,
+static ucs_status_t parse_mem_type_params(const char *optarg,
+                                          ucs_memory_type_t *mem_type)
+{
+    if (!strncmp(optarg, "cuda-managed", 12) &&
```
maybe add a wrapper for `strncmp()` that calculates a length of passed memory type and compare with `optarg`?

dmitrygx commented on src/tools/perf/perftest.c:
```diff
@@ -668,29 +700,20 @@ static ucs_status_t parse_test_params(ucx_perf_params_t *params, char opt, const
-            params->mem_type = UCS_MEMORY_TYPE_ROCM_MANAGED;
-            return UCS_OK;
+        if (UCS_OK != parse_mem_type_params(optarg,
+                                           &params->send_mem_type)) {
```
pls align with the line above

dmitrygx commented on src/tools/perf/perftest.c:
```diff
@@ -668,29 +700,20 @@ static ucs_status_t parse_test_params(ucx_perf_params_t *params, char opt, const
+        optarg2 = strchr(optarg, ',');
+        if (optarg2) {
+            if (UCS_OK != parse_mem_type_params(optarg2 + 1,
+                                               &params->recv_mem_type)) {
```
pls align with the line above

mellanox-github commented:
<details><summary>Mellanox CI: <b>FAILED</b> on 3 of 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 30-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/070aa7ae63c8487c15b5c7e031ac49114475c926/_main) | :x: FAILURE
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/1b4ac405152096c15932aacf756dbd717f86e7a1/hpc-test-node-legacy_W1) | :x: FAILURE
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/208c78f7949f021bfc690f1c0be21619c9822d47/hpc-test-node-legacy_W2) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/00551e54b69268fb582d35df80567dcba66fe764/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/f720435674d4ddf2f819b0063175e169070d9b8e/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/fc3a5121e211dfcec68d5b78b5b174da3c1321f4/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/07f2857cadb83c12fbaacc50b7692d102cbada3c/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/2d0a43b7219fdcf6bfdbae66735e1efcba648769/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/4b28d81ea08292894d2ab83b4214c13b31fc1fa1/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/946a2c0bf9f75c05ff6fe8e88fb85f275181b910/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/94753dcfd4b9cc0b1760d0f4641361af7f5edc8f/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/27acc54514a8bcbefdfefe1a8a3a830fa313b6c3/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/57d128f546594c3d8339b03d3382af42d72d3712/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/9be5fe1cde347cc0140e75d7ef5916f1de21744a/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/ab990456967c82e4f37db6aca00ff0994454bf6b/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/f9c6da81700560ceb87cd2abb19260d6e228d6d7/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/1d1e55944fa42f1a5e81cca03951f3a57bd61cb1/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/ca83dae21bd879e2da2d3bcd3aae0f8af4b78227/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/d39a36998ec9a8d9a91a2a6c0e8ec48f3c0d8d91/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/c5b902d053b4e3a05246bb379e52d6d637bc3b3f/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/60f9c59b84254126dd145ee46a71b111a22dd013/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/fe127565ef0ef10cd0145d000de9ed05032b7c96/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/8418dbefa734d7d319175d49428fe7928bc46950/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/dd3a7aaf90ef59cc6afeec0c33336fddf189927f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e15cecf7d344500c572070e7017f5db5/raw/5624ddb9163100e961ceb1d4661bdfa4d6931190/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

mellanox-github commented:
<details><summary>Mellanox CI: <b>FAILED</b> on 4 of 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 30-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/bdd1023444a34c4c0dca54f459687c5809a785fb/_main) | :x: FAILURE
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/4b7dab616e6fa32cadbd6010ae717d4698f3bba3/hpc-test-node-legacy_W1) | :x: FAILURE
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/c9926cde1175f890e82b5c683ebb0e99f0d70875/hpc-test-node-legacy_W2) | :x: FAILURE
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/808ea7a570faafe38360a07d650660f9f6cf352d/hpc-test-node-legacy_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/ecf96957c7cbeb1e8dfe31d7a5d1e5dfaabf7e58/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/04598953a5a5b5fba8dcf72eca53d4adeb548220/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/e59d710b6f888b5ccb83656b8122ec13939da220/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/a10f5330873a89e76f1c34325cca01244d50aaba/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/6456d4dfcf4fc0a527cb40ee281af52a6cfe7f11/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/7645619cb6cb09e1cbcac499395cc805a84def4d/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/45ee8f4aa1c0081d0b2397be7de509215bc54208/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/7f4271dedc62b791062f4ce0b3466cb6939e2ac2/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/5d01f8c7ca473186d29dce91061d8e0b9c0b08d0/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/611861a71eb33f633e40301d48828b71a34cc886/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/c0fc501166c7569c618d7d4c294e281c21f7c2b3/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/4ce0614c6a500cdb51d8cce5913036cdd86c322d/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/ac825f4ec69ea4b7426930ad28b6854a9eee762e/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/f18b5e0c202b3047a6104a50641671db438ae8ef/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/8a32f7059d728634223de60cc37ed720a59c2c34/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/057065afe195f1fd8326c79b872aaa5e8605f323/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/402ffd22adb38ced06cb9fa2574e5b9a06583744/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/50a6855af5e077e0cf8c987fe935d7e2040227cb/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/9f15a66e1de5bb303ac5ba6bae5c97555a557288/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/9c77b35eacaa5a8627b16c6422ef4087c7e8982d/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a30d504a74297d70042f2c8128b8bffa/raw/07bb11b7ed706bcf5131beae447524fcdebe44f1/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

souravzzz commented:
@dmitrygx I reworked the memory type parsing a little bit so that memory types added in the future will be automatically supported. Also switched to `strcmp` to ensure that typos like `hostx` or `cuda_managed` is not interpreted as `host` or `cuda`.

mellanox-github commented:
<details><summary>Mellanox CI: <b>FAILED</b> on 7 of 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 30-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/989595ff9a824e9ad6a9f8edd411f3c9b6a98dd1/_main) | :x: FAILURE
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/73147c208d1911ea13bcc1a7a3b2148ef64be0d2/hpc-test-node-gpu_W0) | :x: FAILURE
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/ce632e3e62718a70272c68974adda8561e455561/hpc-test-node-legacy_W0) | :x: FAILURE
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/3d1905958748af514f5efc437327145ece7381db/hpc-test-node-legacy_W1) | :x: FAILURE
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/dec2ff25befdf89ba85bfab7e7be6c8dbc91d9ba/hpc-test-node-legacy_W2) | :x: FAILURE
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/c8de25eb2c793fa45dbd1e0901cf68e5841d8794/hpc-test-node-legacy_W3) | :x: FAILURE
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/1c3160304d2935b83b63cd5665aa884004c45428/hpc-test-node-new_W0) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/1778ebeef3bd8a816b73d08816eaa42f58c65d8a/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/f1ca613c90d3ad23625d6cb803cb5ae6630b7a4b/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/79acb6e9649de7db1d44a9c92b4ae9af68a59cbc/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/5fce0ce7cccae9e7980f29897d409ffb9a4638b5/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/5426246b13362405048b1b5832c84c407f5e9fa4/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/7036bf6c77b6bfc24c60d42da70d82d4064aff04/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/ed782655e63fcd1593714d38848b514c8717c02e/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/2dbfe6ff183c63c4e26437a6f05550b18fc80903/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/72e383c74acdfc842e510e7f99265ae5dbb01a26/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/ee3fd5a550d5fd8707ce24841a14f4954dc2c53d/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/a510b42729556432b58472cb529b77c66c00ec8f/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/166ceb8376803b00d340ab6aff5f806876fd46e5/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/f668cd4175a1fd828caf431fa546f59cf7b6a02b/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/554334fa29f1af5b4158b002ae7cdb774710ab1b/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/1cb7455b23f052c1161de722f761f13b059bf9b7/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/9c64cad4cb3c4e2ab981b0482fc4985a225356a9/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/f301c61d27f3b385947604af4226f178f16fae1c/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/e673b8263cf08718de54b373b4461974/raw/2dc8813623655c81ae582dc4efa879b16f21c0ac/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

brminich commented:
the failures are relevant:

```
  CC       ucx_perftest-perftest.o
"/scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node-legacy/worker/0/cont
          rib/../src/tools/perf/perftest.c", line 341: error #188: enumerated
          type mixed with another type
      for (it = 0; it < UCS_MEMORY_TYPE_LAST; it++) {
              ^

"/scrap/jenkins/workspace/hpc-ucx-pr-2/label/hpc-test-node-legacy/worker/0/cont
          rib/../src/tools/perf/perftest.c", line 473: error #188: enumerated
          type mixed with another type
      for (it = 0; it < UCS_MEMORY_TYPE_LAST; it++) {
              ^
```

dmitrygx commented on src/tools/perf/lib/libperf.c:
```diff
@@ -308,9 +308,17 @@ static void ucx_perf_test_prepare_new_run(ucx_perf_context_t *perf,
     perf->params            = *params;
     perf->offset            = 0;
-    perf->allocator         = ucx_perf_mem_type_allocators[params->mem_type];
+    group_index             = rte_call(perf, group_index);
```
please, remove all extra spaces for 313-315:
```
perf->params = *params;
perf->offset = 0;
group_index  = rte_call(perf, group_index);
```

dmitrygx commented on src/tools/perf/lib/libperf.c:
```diff
@@ -1573,8 +1597,9 @@ ucs_status_t ucx_perf_run(ucx_perf_params_t *params, ucx_perf_result_t *result)
     if (perf->allocator == NULL) {
-        ucs_error("Unsupported memory type %s",
-                  ucs_memory_type_names[params->mem_type]);
+        ucs_error("Unsupported memory types %s,%s",
```
```
"Unsupported memory types %s<->%s"
```

dmitrygx commented on src/tools/perf/perftest.c:
```diff
@@ -365,20 +382,9 @@ static void usage(const struct perftest_context *ctx, const char *program)
-        printf("                        rocm-managed - AMD/ROCm GPU managed memory\n");
-    }
+    printf("     -m <mem type>[,<mem type>]\n");
+    printf("                    memory type of message for sender and receiver (host)\n");
```
what "(host)" mean here? default value for send/recv mem_types?

souravzzz replied:
Yes, (this) is how default values are currently shown for other options

dmitrygx commented on src/tools/perf/perftest.c:
```diff
@@ -460,6 +466,41 @@ static ucs_status_t parse_ucp_datatype_params(const char *optarg,
+                                          ucs_memory_type_t *recv_mem_type)
+{
+    const char *delim = ",";
+    char *token = strtok((char*)optarg, delim);
```
```
const char *delim = ",";
char *token       = strtok((char*)optarg, delim);
```

mellanox-github commented:
<details><summary>Mellanox CI: <b>FAILED</b> on 2 of 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 09-Jan-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/6897aa0ae7e7774b566e1d0b73bd721e19ecb25d/_main) | :x: FAILURE
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/39b4fafa4b120ccb8ac505a3610875d715e39414/hpc-test-node-gpu_W3) | :x: FAILURE
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/388099fa7ecd35e6f4d3bc2a7a2c6add953b35ce/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/9118144997e57e8678e376829e2632aaad230cdc/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/d01ab1b1364d2b014e352fc46218884faadf9fca/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/b687d41d2513e18ed67d85881e8a358f8556a053/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/9aaf7e882aaba0179561a0eb9e5c3e39f57f9e6f/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/53eb2453b23cc234eaa6881dc663a4582892c8eb/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/fa104b9a120274405c536edd2a818ce1caf6df03/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/e4256b5c3615890e0dcf4bcf45b8ece067f3c44e/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/233c8e3deffed4f7bde06608321f263ce2bf2656/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/4f7182ab4938d0fa6cf0bdaf83a7ac39a5937db8/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/3a6f9fa79c7cf7271b112f37faba52eb83250178/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/c835b6c1f64d0808954f5b4ef655133e5d11e06a/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/1adf75d5f8f43445cfa71814169447564faaaee8/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/c9e8a4b6b9634acaadbe1740a7acb6f97175d285/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/005ea47a2feca8bc2469584f644dfe2002c9bdf1/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/bc7fcd0667fe8b076fb893d30f33bb921b778bc9/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/ef9ebb261859b88df1f8af1093a47219883b8588/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/891e9048949b16826c6c5252741f997139d1c1aa/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/9cea22d935bfe959f1635c0e1d8c044dae4b8676/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/99b3e324364afb208e3da6bb067a4e81465d963d/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/a8711cb444d4bc747b8d99ab65e2b9173d5962b2/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/b77198a13ebbca1fbbc9f15d4c3e8e6aa484e860/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/9f9f0f8d875bdc47f35ef0c7b351419c/raw/c662e00b144e2e7b45b02b33eb0b6b474cb4d028/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 09-Jan-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/60017459207dedb467dcaffc2a45bcb1e0822b96/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/e0a6da79357ca5275ba52b46c43ca87ac7674b59/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/195a8bc9f735a294b8c9b4820d64af14e52f55e8/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/cf1d40b8f8c29f8c4c5e813805de8bf13eea7cd3/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/96e2338bd763899a7ddeb10afbc1384e4d097731/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/114bd644198fe548010f7532598c1782d1ec0971/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/b628b85f3be580e45d9f05e6f9d5d32b84abf10f/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/65a15be0c38ce2d443c03816c427d304c0fea06c/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/ff4d3af81bdf69c77cc93f73084322ec2fac6649/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/3cd3b98a76aa27c07c180f36d39e16e0144b7f94/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/8846c294f92b454c580e5bb8eda76f12e8da1b13/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/3f89707cc9cfd0bffb23ffa2c43c37a51e56aecd/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/8cd17a6e2e3be380e0aed6dee17458e946b11ac3/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/7d8102f02c6b693ea52deb6dfcf837564bd6ea7e/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/ff5b4da3c6ab49f20dbcaf791b5846847060b0d4/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/693dfe5e72201e3e390938385cd8e3cd43d847fe/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/e134c3bc4816eef9f97567d8d08ceddac0bfaf41/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/69358936bfb834bca39ca06227406358934e4bd9/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/1be11a6624a7dac2fd39e71f9cff716ca79a2959/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/a466fcd9e836a7eff5d7446a983b569e7612e566/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/41b247c05971d4172a9eeb135fbeeba8fdd63f88/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/7ff3a9f3c56413c3d2b74ea4ff738bbe5e42e255/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/d51eae311f0cd709678889674cdbf2c5c336a902/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/5ddcc44d919e4fd14038d8dff8efb61f33f37243/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/096b49842cf2fae41bffd6049a5195fd/raw/bb9b5a3c55b5ac93d5ed88030cc6f06fc90b9d5f/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

souravzzz commented:
bot:pipe:retest

dmitrygx commented:
@souravzzz the failure is unrelated to your PR
we are waiting for #4640 that should fix the issue
after that, we will restart the pipeline CI for your PR

dmitrygx commented:
bot:pipe:retest

brminich commented:
@alinask, plz review as well

alinask commented on src/tools/perf/perftest.c:
```diff
@@ -365,20 +379,9 @@ static void usage(const struct perftest_context *ctx, const char *program)
-    if (ucx_perf_mem_type_allocators[UCS_MEMORY_TYPE_ROCM_MANAGED] != NULL) {
-        printf("                        rocm-managed - AMD/ROCm GPU managed memory\n");
-    }
+    printf("     -m <mem type>[,<mem type>]\n");
```
@souravzzz whose mem type goes first if the types are different? 
Could you please document the usage in this case?

mellanox-github commented:
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 14-Jan-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/dc26a750ddf0ff72f922c6828635d0e2a2f30d7a/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/9d71f37e91aef311a3c0b1e967a927f65329a603/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/9ddc8aa6cfc4783b600664bb75731164503b6245/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/fda861985f3f1460d7a41676af6088c39c6625ca/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/663477040fdb9f353b7069ce8ccec6feb00df794/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/72116c141621329a72af0583aafb758d3110fff1/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/41f950df383b1b3d4b7c69bf7c73f815f0d62f6f/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/cbe05ca9dbdffbdb161f77ef9eb2e106c79d8c02/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/37f8b93265c9356780eeb68f9d617f0dcd5cb5a0/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/c5a82de3a28f5097728b36594a660d21826e03d2/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/a20d3c51b44b2e6e4b4f4242e0a327eae3c00be6/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/64d1ccfb9aa439c9c673932ab17d180ff7c2c2c6/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/bb563294e813cff651a974f1de070c90e4a5adc5/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/692a95ac2e46207f7a64002fb2d94b25cbdab678/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/ff5cd4038ddb70316ee7cb7fbaaf17406aa70027/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/d6d9bf2619f3e5936ce4c75dfbc2b0bd6164fbcc/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/82b3a914f6b5e4fd4d0b1d6b24c07a26b1eab530/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/0264ce0aa27cd1c2c9fb8a28f621d4a69b1acc2d/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/5d8c96e2daa68b47ef153e9c195ef3c7e5fe9edd/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/0268d2de7e44d15148ecbc7d463a7b54da09b355/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/a8cd4bd543373c22dc87a76636bb56b0e099da5a/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/1e32e4f9c579acbd7c74743392f0f129e0849b11/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/ceafb7b3a8c9008dcf8cddd8735ab4aed6a8cb68/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/2d99d4ac51f91ac23138a7a2d01ad2226610d449/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/afb4951812527a29e46b566ac5d914eb/raw/255c65666894cdf3b0e63990277e60ed37f23357/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS
